### ISSUE TITLE
EISDIR & absolute path errors

### ISSUE THREAD

**Issue by geneh** [clearlydefined/crawler#89]
The following Maven components have ended up in dead letter queue:
```
1 'clearlydefined' 'cd:/sourcearchive/mavencentral/org.scala-lang/scala-reflect/2.13.0-M3-f73b161' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-Ht1z4x/scala/reflect/\''
2 'clearlydefined' 'cd:/sourcearchive/mavencentral/org.scala-lang/scala-library/2.13.0-M3-f73b161' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-BhEZPG/scala/annotation/\''
3 'clearlydefined' 'cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-pdf-itext5/9.1.12' 'absolute path: /'
4 'clearlydefined' 'cd:/sourcearchive/mavencentral/org.jsonschema2pojo/jsonschema2pojo-cli/1.0.0-alpha2' 'absolute path: /'
5 'clearlydefined' 'cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded-query/9.1.7.Final' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-7dkcT0/org/infinispan/\''
6 'clearlydefined' 'cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-swt/9.1.12' 'absolute path: /'
7 'clearlydefined' 'cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-log4j/9.1.12' 'absolute path: /'
8 'clearlydefined' 'cd:/sourcearchive/mavencentral/com.facebook.presto/presto-jdbc/0.197' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-1AWdll/com/facebook/\''
9 'clearlydefined' 'cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded/9.1.7.Final' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-GXCWCD/org/infinispan/\''
```

```
17 'scancode' 'cd:/sourcearchive/mavencentral/xerces/xercesImpl/2.11.0' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-y14wKn/org/apache/\''
18 'scancode' 'cd:/sourcearchive/mavencentral/postgresql/postgresql/9.1-901-1.jdbc4' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-a5CnLp/org/postgresql/\''
19 'scancode' 'cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-swt/9.1.12' 'absolute path: /'
20 'scancode' 'cd:/sourcearchive/mavencentral/com.github.haifengl/smile-plot/1.5.1' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-VMQ5pt/smile/plot/\''
21 'scancode' 'cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-pdf-itext5/9.1.12' 'absolute path: /'
22 'scancode' 'cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-log4j/9.1.12' 'absolute path: /'
23 'scancode' 'cd:/sourcearchive/mavencentral/com.facebook.swift/swift-codec/0.23.1' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-3c0vcO/com/facebook/\''
24 'scancode' 'cd:/sourcearchive/mavencentral/org.robovm/robovm-objc/1.14.0' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-7tNmKz/META-INF/maven/org.robovm/\''
25 'scancode' 'cd:/sourcearchive/mavencentral/org.eclipse.collections/eclipse-collections-testutils/9.2.0.M1' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-joOnc7/META-INF/maven/org.eclipse.collections/\''
26 'scancode' 'cd:/sourcearchive/mavencentral/org.parboiled/parboiled-java/1.1.8' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-0AAPft/org/parboiled/\''
27 'scancode' 'cd:/sourcearchive/mavencentral/org.scala-lang/scala-library/2.13.0-M3-f73b161' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-1RDHCQ/scala/annotation/\''
28 'scancode' 'cd:/sourcearchive/mavencentral/com.github.haifengl/smile-core/1.5.1' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-XkWZLj/smile/association/\''
29 'scancode' 'cd:/sourcearchive/mavencentral/com.facebook.presto/presto-jdbc/0.197' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-2n9DO2/com/facebook/\''
30 'scancode' 'cd:/sourcearchive/mavencentral/org.robovm/robovm-cocoatouch/1.14.0' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-uwpvfg/META-INF/maven/org.robovm/\''
31 'scancode' 'cd:/sourcearchive/mavencentral/org.easymock/easymock/3.5.1' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-nlept5/org/easymock/\''
32 'scancode' 'cd:/sourcearchive/mavencentral/org.scala-lang/scala-reflect/2.13.0-M3-f73b161' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-XndU5Y/scala/reflect/\''
33 'scancode' 'cd:/sourcearchive/mavencentral/org.robovm/robovm-compiler/1.12.0' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-VL5iEP/META-INF/maven/org.robovm/\''
34 'scancode' 'cd:/sourcearchive/mavencentral/cglib/cglib-sample/3.2.6' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-JYhO13/net/sf/\''
35 'scancode' 'cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded-query/9.1.7.Final' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-4uvfqA/org/infinispan/\''
36 'scancode' 'cd:/sourcearchive/mavencentral/com.twitter/chill-java/0.9.2' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-p4rhhS/com/twitter/\''
37 'scancode' 'cd:/sourcearchive/mavencentral/im.actor/actor-botkit/1.0.113' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-H7MaaB/im/actor/\''
38 'scancode' 'cd:/sourcearchive/mavencentral/org.mp4parser/isoparser/1.9.31' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-Vrb1bu/org/mp4parser/\''
39 'scancode' 'cd:/sourcearchive/mavencentral/com.github.haifengl/smile-data/1.5.1' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-tdcMi6/smile/data/\''
40 'scancode' 'cd:/sourcearchive/mavencentral/org.jsonschema2pojo/jsonschema2pojo-cli/1.0.0-alpha2' 'absolute path: /'
41 'scancode' 'cd:/sourcearchive/mavencentral/com.github.haifengl/smile-math/1.5.1' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-4rv707/smile/hash/\''
42 'scancode' 'cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded/9.1.7.Final' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-oonU3g/org/infinispan/\''
43 'scancode' 'cd:/sourcearchive/mavencentral/com.twitter/parquet-jackson/1.6.0' 'EISDIR: illegal operation on a directory, open \'/tmp/cd-YODXbh/parquet/org/\''
```

Full message examples:
```
{"type":"clearlydefined","url":"cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded-query/9.1.7.Final","context":{"history":["package@cd:/maven/mavencentral/org.infinispan/infinispan-embedded-query","maven@cd:/maven/mavencentral/org.infinispan/infinispan-embedded-query","source@cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded-query/9.1.7.Final","clearlydefined@cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded-query/9.1.7.Final"],"qualifier":"urn:"},"policy":{"fetch":"mutables","freshness":"match","map":{"name":"package","path":"/maven/source/clearlydefined"}},"_metadata":{"version":1,"meta":{"loopName":"0","cid":"54g","root":"package@cd:/maven/mavencentral/org.infinispan/infinispan-embedded-query"},"type":"deadletter","url":"cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded-query/9.1.7.Final","processedAt":"2018-03-23T07:55:22.415Z","fetchedAt":"2018-03-23T07:55:22.415Z","links":{"self":{"href":"urn:deadletter:c3c5474f-bce1-4c3d-8e18-819672e34510","type":"resource"}},"extra":{"type":"clearlydefined","url":"cd:/sourcearchive/mavencentral/org.infinispan/infinispan-embedded-query/9.1.7.Final","reason":"EISDIR: illegal operation on a directory, open '/tmp/cd-7dkcT0/org/infinispan/'"}}}
```

```
{"type":"clearlydefined","url":"cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-swt/9.1.12","context":{"history":["package@cd:/maven/mavencentral/org.xhtmlrenderer/flying-saucer-swt","maven@cd:/maven/mavencentral/org.xhtmlrenderer/flying-saucer-swt","source@cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-swt/9.1.12","clearlydefined@cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-swt/9.1.12"],"qualifier":"urn:"},"policy":{"fetch":"mutables","freshness":"match","map":{"name":"package","path":"/maven/source/clearlydefined"}},"_metadata":{"version":1,"meta":{"loopName":"1","cid":"4u6","root":"package@cd:/maven/mavencentral/org.xhtmlrenderer/flying-saucer-swt"},"type":"deadletter","url":"cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-swt/9.1.12","processedAt":"2018-03-23T06:46:13.221Z","fetchedAt":"2018-03-23T06:46:13.221Z","links":{"self":{"href":"urn:deadletter:8bd4bbde-275d-4907-a149-46411f0debd5","type":"resource"}},"extra":{"type":"clearlydefined","url":"cd:/sourcearchive/mavencentral/org.xhtmlrenderer/flying-saucer-swt/9.1.12","reason":"absolute path: /"}}}
```

**Comment by geneh** [comment]
The error is reproducible when re-queued.

**Comment by geneh** [comment]
`absolute path: /` error seems like a real error, so won't fix.

**Comment by geneh** [comment]
This error should either be fixed or another library be found that does not have this problem.

**Comment by geneh** [comment]
https://search.maven.org/remotecontent?filepath=com/twitter/parquet-jackson/1.6.0/parquet-jackson-1.6.0-sources.jar

**Comment by jeffmcaffer** [comment]
@geneh, what is the status of this? Still an issue?

**Comment by geneh** [comment]
Yes, there is a problem with `decompress` package.
Example of error repro:
```
{
  "type": "scancode",
  "url": "cd:/sourcearchive/mavencentral/com.facebook.presto/presto-jdbc/0.205"
}
```
BaseHandler.decompress() fails with: `Error: ENOENT: no such file or directory, open 'cd-g29Co3/com/facebook/'`

**Comment by dabutvin** [comment]
fixed in #176
